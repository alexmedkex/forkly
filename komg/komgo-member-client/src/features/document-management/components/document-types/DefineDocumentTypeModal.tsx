import * as React from 'react'
import { Form, Modal, Button, Input, Dropdown } from 'semantic-ui-react'
import { Category } from '../../store/types'

export interface Props {
  visible: boolean
  title: string
  categories: Category[]
  predefinedData: { category: string; name: string; id?: string }
  toggleVisible(): void
  onCreateSuccess(cat: string, name: string): void
  onEditSuccess(cat: string, name: string, id: string): void
}

interface State {
  category: string
  name: string
}

interface CategoryRepresentation {
  key: string
}

const formatCategoriesToSelect = (categories: Category[]): CategoryRepresentation[] => {
  return categories.map((cat: Category) => {
    return { key: cat.id, text: cat.name, value: cat.id }
  })
}

class DefineDocumentTypeModal extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props)
  }

  handleSubmit = () => {
    if (this.props.predefinedData.id) {
      this.props.onEditSuccess(
        this.state.category ? this.state.category : this.props.predefinedData.category,
        this.state.name,
        this.props.predefinedData.id
      )
    } else {
      this.props.onCreateSuccess(this.state.category, this.state.name)
    }
  }

  render() {
    return (
      <Modal open={this.props.visible} size="mini">
        <Modal.Header>{this.props.title}</Modal.Header>
        <Modal.Content>
          <Modal.Description>
            <Form onSubmit={this.handleSubmit} id="submit-form">
              <Form.Field>
                <label>Category</label>
                <Dropdown
                  placeholder="Select category"
                  defaultValue={
                    this.props.predefinedData.category !== '' ? this.props.predefinedData.category : undefined
                  }
                  options={formatCategoriesToSelect(this.props.categories)}
                  onChange={this.setSelectedCategoryId}
                />
              </Form.Field>
              <Form.Field>
                <label>Name</label>
                <Input
                  placeholder="My new doctype"
                  defaultValue={this.props.predefinedData.name !== '' ? this.props.predefinedData.name : undefined}
                  onChange={(e: any) => {
                    this.setState({ name: e.target.value })
                  }}
                  size="mini"
                />
              </Form.Field>
            </Form>
          </Modal.Description>
        </Modal.Content>
        <Modal.Actions>
          <Button className="ui negative button" role="button" onClick={this.props.toggleVisible}>
            Cancel
          </Button>
          <Button primary={true} content="Submit" form="submit-form" />
        </Modal.Actions>
      </Modal>
    )
  }

  private setSelectedCategoryId = (e: any, { value }: { value: string }) => {
    this.setState({ category: value })
  }
}

export default DefineDocumentTypeModal
