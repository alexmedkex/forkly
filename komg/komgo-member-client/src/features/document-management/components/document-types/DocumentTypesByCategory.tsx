import * as React from 'react'
import { List, Accordion, AccordionPanelProps, Checkbox, Icon } from 'semantic-ui-react'
import styled from 'styled-components'
import { indigo, darkViolet } from '../../../../styles/colors'

import { DocumentType, Category } from '../../../document-management/store/types'
import { groupBy } from '../../../document-management/components/documents/my-documents/toMap'

import { PanelCheckboxState } from '../templates/DocumentRequestModal'

export interface Props {
  documentTypes: DocumentType[]
  categories: Category[]
  selectedDocumentTypesByCategory: Map<string, DocumentType[]>
  panelCheckboxState: Map<string, PanelCheckboxState>
  markAsDisabled: (documentTypeName: string) => { isDisabled: boolean; disabledText: string }
  isSelected(documentType: DocumentType): boolean
  onPanelClick(category: Category): void
  onListItemClick(documentType: DocumentType): void
}

class DocumentTypesByCategory extends React.Component<Props> {
  render() {
    return (
      <StyledAccordion
        exclusive={false}
        fluid={true}
        panels={this.props.categories.map(this.categoryToAccordionPanel)}
      />
    )
  }

  private categoryToAccordionPanel = (category: Category): AccordionPanelProps => {
    return {
      key: category.id,
      name: category.name,
      title: { content: this.categoryToPanelTitle(category) },
      content: { content: this.categoryToList(category) }
    }
  }

  private categoryToPanelTitle = (category: Category) => {
    return (
      <>
        <StyledIcon name="chevron down" />
        <StyledCheckbox
          key={category.id}
          floated="left"
          label={category.name}
          checked={this.props.panelCheckboxState.get(category.id) === 1}
          indeterminate={this.props.panelCheckboxState.get(category.id) === 0}
          onClick={(e: React.MouseEvent) => {
            e.preventDefault()
            e.stopPropagation()
            this.props.onPanelClick(category)
          }}
        />
      </>
    )
  }

  private categoryToList = (category: Category) => {
    const groupByCategoryId = (docType: DocumentType) => docType.category.id
    const documentTypesForThisCategory = groupBy(this.props.documentTypes, groupByCategoryId).get(category.id)
    if (!(documentTypesForThisCategory && documentTypesForThisCategory.length)) {
      return null
    }

    return (
      <List
        selection={true}
        name={<b>category.name</b>}
        items={documentTypesForThisCategory.map(this.documentTypeToListItem)}
      />
    )
  }

  private documentTypeToListItem = (documentType: DocumentType) => {
    // default values
    let disabledText: string = ''
    let isDisabled: boolean = false

    // override defaults if markAsDisabled function is provided
    if (this.props.markAsDisabled) {
      const result = this.props.markAsDisabled(documentType.name)
      if (result.isDisabled) {
        disabledText = result.disabledText
        isDisabled = result.isDisabled
      }
    }
    return (
      <List.Item style={{ padding: '7px 2.5em' }}>
        <List.Content key={documentType.id} floated="left">
          <Checkbox
            disabled={false} // We are allowing to re-request same document several times
            label={documentType.name}
            onClick={() => this.props.onListItemClick(documentType)}
            checked={this.props.isSelected(documentType)}
          />
        </List.Content>
        <List.Content key={documentType.id} floated="right">
          <p className="disabled" hidden={!isDisabled}>
            {disabledText}
          </p>
        </List.Content>
      </List.Item>
    )
  }
}

export const StyledAccordion = styled(Accordion)`
  &&& {
    .komgo-accordion-title {
      background-color: ${indigo} !important;
      border-radius: 5px;
      border: 1px solid ${darkViolet};
      padding: 12px 30px;
      height: 45px;
      color: white;
    }

    .title .dropdown.icon {
      display: none;
    }

    .active .komgo-accordion-title {
      border-radius: 5px 5px 0px 0px;
      margin-bottom: -14px;
    }

    i {
      color: white;
      margin-top: 2px;
    }
  }
`

const StyledCheckbox = styled(Checkbox)`
  &&& {
    padding-left: 1em;
    label {
      font-weight: bold;
    }
    .dropdown.icon {
      display: none;
    }
  }
`

const StyledIcon = styled(Icon)`
  &&& {
    width: 16px;
  }
`

export default DocumentTypesByCategory
