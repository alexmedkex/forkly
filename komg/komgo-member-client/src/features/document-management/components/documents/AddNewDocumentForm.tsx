import { Profile } from '../../../../store/common/types'
import { Document } from '../../../../features/document-management/store'
import { ApplicationState } from '../../../../store/reducers'
import { Formik, FormikProps } from 'formik'
import * as React from 'react'
import { connect } from 'react-redux'
import { Form } from 'semantic-ui-react'
import * as Yup from 'yup'
import * as path from 'path'

import { Category, CreateDocumentRequest, DocumentType } from '../../store/types'
import { categoryIdValidation, DocumentCategoryField } from './new-document-components/DocumentCategoryField'
import { DocumentNameField, nameValidation } from './new-document-components/DocumentNameField'
import { DocumentTypeField, typeIdValidation } from './new-document-components/DocumentTypeField'
import { FileUploadField, fileValidation } from './new-document-components/FileUploadField'
import { sortCategories, sortDocumentTypes } from '../../utils/sortingHelper'

export interface Props {
  documents: Document[]
  categories?: Category[]
  documentTypes: DocumentType[]
  profile: Profile
  preselectedCategory: string
  preselectedDocumentType: string
  documentTypeDisabled?: boolean
  handleSubmit(formData: CreateDocumentRequest): void
  toggleSubmit?(enable: boolean): void
}

interface State {
  fileName: string
}

interface NewDocumentForm {
  name: string
  categoryId?: string
  typeId: string
  file: File | null
  fileName: string
}

export class AddNewDocumentForm extends React.Component<Props, State> {
  private validationSchema

  constructor(props: Props) {
    super(props)

    this.validationSchema = this.formValidationSchema()

    this.onFileNameUpdate = this.onFileNameUpdate.bind(this)
  }

  render() {
    return (
      <Formik
        initialValues={{
          name: '',
          categoryId: this.props.preselectedCategory || undefined,
          typeId:
            this.getPreselectedDocumentType(
              this.props.preselectedDocumentType,
              this.props.documentTypes,
              this.props.preselectedCategory
            ) || '',
          file: null,
          fileName: ''
        }}
        validationSchema={this.validationSchema}
        onSubmit={(values, actions) => {
          this.props.handleSubmit({
            name: values.name,
            categoryId: values.categoryId,
            documentTypeId: values.typeId,
            file: values.file!,
            creator: {
              firstName: this.props.profile.firstName,
              lastName: this.props.profile.lastName,
              companyId: this.props.profile.company
            }
          })
        }}
        render={(props: FormikProps<NewDocumentForm>) => {
          const { values, handleSubmit } = props

          const categoryId = this.props.categories ? values.categoryId : undefined
          this.props.toggleSubmit(props.isValid) // TODO: this logic should be refactored and moved from render

          return (
            <Form onSubmit={handleSubmit} id="submit-form" className="add-new-document">
              <DocumentNameField formik={props} />

              {this.props.categories && (
                <DocumentCategoryField
                  formik={props}
                  categories={sortCategories(this.props.categories)}
                  preselectedCategory={this.props.preselectedCategory}
                />
              )}

              <DocumentTypeField
                formik={props}
                categoryId={categoryId}
                documentTypes={sortDocumentTypes(this.props.documentTypes)}
                preselectedDocumentType={this.getPreselectedDocumentType(
                  this.props.preselectedDocumentType,
                  this.props.documentTypes,
                  values.categoryId
                )}
                disabled={this.props.documentTypeDisabled}
              />

              <FileUploadField formik={props} onFileNameUpdate={this.onFileNameUpdate} />
            </Form>
          )
        }}
      />
    )
  }

  /**
   * A document type is preselected if:
   * - It is passed explicitly as a prop or if it is.
   * - It is the only document type available for that category, in this case it is also preselected and disabled.
   */
  getPreselectedDocumentType(
    preselectedDocumentType: string,
    documentTypes: DocumentType[],
    categoryId: string
  ): string {
    const filteredDocumentTypes: DocumentType[] = documentTypes.filter(x => x.category.id === categoryId)
    if (preselectedDocumentType) {
      return preselectedDocumentType
    } else if (filteredDocumentTypes.length === 1) {
      return filteredDocumentTypes[0].id
    } else {
      return ''
    }
  }

  // To get document extension to build full name
  onFileNameUpdate(fileName: string) {
    this.setState({
      fileName
    })
  }

  private formValidationSchema(): any {
    const fields = {
      name: this.validateDocumentName(),
      ...typeIdValidation,
      ...fileValidation
    }

    return this.props.categories
      ? Yup.object().shape({
          ...fields,
          ...categoryIdValidation
        })
      : Yup.object().shape(fields)
  }

  private validateDocumentName() {
    return Yup.string()
      .required('Document title is required')
      .test(
        'unique-name',
        'A document with the same name already exists. Choose another name.',
        this.noDocumentWithTheSameName.bind(this)
      )
  }

  private noDocumentWithTheSameName(name: string): boolean {
    if (!this.state || !this.state.fileName) {
      return true
    }

    const docNameWithExt = `${name}${path.extname(this.state.fileName)}`
    const foundDocument = this.props.documents.find(doc => doc.name === docNameWithExt)
    return !foundDocument
  }
}

const mapStateToProps = (state: ApplicationState) => ({
  profile: state.get('uiState').get('profile')
})

export default connect(mapStateToProps, {})(AddNewDocumentForm)
