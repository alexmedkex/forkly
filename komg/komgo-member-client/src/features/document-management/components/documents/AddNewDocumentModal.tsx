import * as React from 'react'
import { Modal, Button } from 'semantic-ui-react'

interface Props {
  visible: boolean
  title: string
  children: any
  toggleVisible(): void
}

interface State {
  submitButtonDisabled: boolean
}

class AddNewDocumentModal extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props)
    this.state = { submitButtonDisabled: true }
    this.toggleSubmitButton = this.toggleSubmitButton.bind(this)
  }

  toggleSubmitButton(option: boolean) {
    if (!option !== this.state.submitButtonDisabled) {
      this.setState({ submitButtonDisabled: !option })
    }
  }

  render() {
    return (
      <Modal open={this.props.visible} centered={true} style={{ top: 'unset' }}>
        <Modal.Header>{this.props.title}</Modal.Header>
        <Modal.Content>
          {React.Children.map(this.props.children, child => {
            return React.cloneElement(child, {
              // This code injects the toggleSubmit handler into the this.props.children
              toggleSubmit: this.toggleSubmitButton
            })
          })}
        </Modal.Content>
        <Modal.Actions>
          <Button className="ui button" role="button" onClick={this.props.toggleVisible}>
            Cancel
          </Button>
          <Button disabled={this.state.submitButtonDisabled} primary={true} content="Add document" form="submit-form" />
        </Modal.Actions>
      </Modal>
    )
  }
}

export default AddNewDocumentModal
