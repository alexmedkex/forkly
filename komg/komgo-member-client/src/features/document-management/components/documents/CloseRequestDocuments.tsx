import * as React from 'react'
import { Confirm } from 'semantic-ui-react'

interface Props {
  open: boolean
  content: string
  onConfirmClose(): void
  onToggleVisible(): void
}

class CloseRequestDocuments extends React.Component<Props> {
  handleConfirm = () => {
    this.props.onConfirmClose()
  }

  renderContent = (content: string) => {
    return { content }
  }

  render() {
    const { onToggleVisible, open, content } = this.props
    return (
      <Confirm
        open={open}
        cancelButton="Cancel"
        confirmButton="Confirm"
        header="Close request"
        content={this.renderContent(content)}
        onCancel={() => onToggleVisible()}
        onConfirm={() => this.handleConfirm()}
      />
    )
  }
}

export default CloseRequestDocuments
