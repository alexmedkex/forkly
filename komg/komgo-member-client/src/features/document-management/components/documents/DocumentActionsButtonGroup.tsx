import * as React from 'react'
import { Button, Container } from 'semantic-ui-react'
import styled from 'styled-components'
interface Props {
  visible: boolean
  counterparty?: boolean
  userCanCrudAndShareDocs?: boolean
  toggleShareDocumentModalVisible?: () => void
  downloadSelectedDocuments: () => void
}

class DocumentActionsButtonGroup extends React.Component<Props> {
  constructor(props: Props) {
    super(props)
  }

  render() {
    const { visible, toggleShareDocumentModalVisible } = this.props
    if (this.props.counterparty) {
      return (
        <ContainerWithMargin visible={visible ? 1 : 0}>
          <Button.Group>
            <Button onClick={() => this.props.downloadSelectedDocuments()}>Download</Button>
          </Button.Group>
        </ContainerWithMargin>
      )
    } else {
      return (
        <ContainerWithMargin visible={visible ? 1 : 0}>
          <Button.Group>
            <ShareButton
              userCanCrudAndShareDocs={!!this.props.userCanCrudAndShareDocs}
              toggleShareDocumentModalVisible={toggleShareDocumentModalVisible}
            />
            {/* <Button disabled={true}>Revoke</Button> */}
            <Button onClick={() => this.props.downloadSelectedDocuments()}>Download</Button>
          </Button.Group>
        </ContainerWithMargin>
      )
    }
  }
}

const ShareButton = (props: { userCanCrudAndShareDocs: boolean; toggleShareDocumentModalVisible?: () => void }) => {
  if (props.userCanCrudAndShareDocs) {
    return (
      <Button onClick={() => props.toggleShareDocumentModalVisible && props.toggleShareDocumentModalVisible()}>
        Share
      </Button>
    )
  } else {
    return null
  }
}

const ContainerWithMargin = styled(Container)`
  margin-top: 0.5rem;
  margin-bottom: -0.14285714em;
  visibility: ${({ visible }) => (visible ? 'visible' : 'hidden')};
`

export default DocumentActionsButtonGroup
