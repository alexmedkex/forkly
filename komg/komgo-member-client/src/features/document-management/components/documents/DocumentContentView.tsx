import { ErrorMessage } from '../../../../components'
import { PERMITTED_MIME_TYPES } from '../../../document-management/utils/permittedMimeTypes'
import * as React from 'react'
import { Dimmer, Loader, Image } from 'semantic-ui-react'

import PdfContentView from './PdfContentView'

interface Props {
  documentContent: string
  documentType: string
  isLoadingContent: boolean
}

const DocumentContentView: React.SFC<Props> = (props: Props) => {
  if (props.isLoadingContent) {
    return (
      <Dimmer active={true}>
        <Loader indeterminate={true}>Loading</Loader>
      </Dimmer>
    )
  }

  if (!Object.values(PERMITTED_MIME_TYPES).includes(props.documentType)) {
    return <ErrorMessage title="Cannot view document" error="Document type not permitted" />
  }

  if (props.documentType === PERMITTED_MIME_TYPES.PDF_MIME_TYPE) {
    return <PdfContentView documentContent={props.documentContent} />
  }
  if (
    props.documentType === PERMITTED_MIME_TYPES.JPEG_MIME_TYPE ||
    props.documentType === PERMITTED_MIME_TYPES.PNG_MIME_TYPE
  ) {
    return <Image src={`data:${props.documentType};base64, ${props.documentContent}`} />
  }
  return <iframe height="500em" width="100%" src={`data:${props.documentType};base64, ${props.documentContent}`} />
}

export default DocumentContentView
