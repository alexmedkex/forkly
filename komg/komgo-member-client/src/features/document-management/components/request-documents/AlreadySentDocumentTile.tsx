import React, { useState } from 'react'
import { Document } from '../../store'
import styled from 'styled-components'
import SpanAsLink from '../../../../components/span-as-link/SpanAsLink'
import { printAlreadySharedDocumentDate } from '../../../../../src/utils/timer'
import moment from 'moment-timezone'

interface IProps {
  counterpartyId: string
  document: Document
  index: number
  openViewDocument(previewDocumentId: string): void
}

export const AlreadySentDocumentTile: React.FC<IProps> = (props: IProps) => {
  return (
    <DocumentStyle>
      <StyledIndex data-test-id={`already-sent-doc-index-${props.document.id}`}>{props.index}</StyledIndex>
      <DocumentFrame>
        <ElementStyle>
          <SpanAsLink
            data-test-id={`already-sent-doc-link-${props.document.id}`}
            onClick={() => props.openViewDocument(props.document.id)}
          >
            {props.document.name}
          </SpanAsLink>
        </ElementStyle>
        <ElementRightStyle />
        <ElementRightStyle data-test-id={`already-sent-doc-date-${props.document.id}`}>
          {printAlreadySharedDocumentDate(
            moment(
              props.document.sharedWith.find(shared => shared.counterpartyId === props.counterpartyId).sharedDates[0]
            )
          )}
        </ElementRightStyle>
      </DocumentFrame>
    </DocumentStyle>
  )
}

const StyledIndex = styled.div`
  height: 21px;
  width: 5px;
  color: #5d768f;
  font-family: Helvetica;
  font-size: 14px;
  line-height: 21px;
`

const ElementStyle = styled.div`
  margin: 5px;
`

const ElementRightStyle = styled.div`
  margin: 5px;
  text-align: right;
`

const DocumentStyle = styled.div`
  padding-left: 15px;
  display: grid;
  grid-template-columns: 5px [index] auto [panel-item];
  align-items: center;
`

const DocumentFrame = styled.div`
  display: grid;
  grid-template-columns: auto [panel-item] auto [panel-item] auto [panel-item];
  border: 1px solid #e8eef3;
  border-radius: 4px;
  margin: 5px 5px 5px 15px;
  padding-left: 12px;
`
