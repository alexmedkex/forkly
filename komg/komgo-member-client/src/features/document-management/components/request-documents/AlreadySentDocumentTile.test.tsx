import { shallow } from 'enzyme'
import * as React from 'react'
import * as renderer from 'react-test-renderer'
import { mockDocumentTypes } from '../../store/document-types/mock-data'
import { mockCategories } from '../../store/categories/mock-data'
import { mockDocuments } from '../../store/documents/mock-data'
import { AlreadySentDocumentTile } from './AlreadySentDocumentTile'

describe('AlreadySentDocumentTile', () => {
  const defaultProps = {
    document: mockDocuments[0],
    index: 20,
    openViewDocument: jest.fn(),
    counterpartyId: 'societegenerale'
  }

  it('should match default snapshot', () => {
    expect(renderer.create(<AlreadySentDocumentTile {...defaultProps} />).toJSON()).toMatchSnapshot()
  })

  it('should find the index', () => {
    const wrapper = shallow(<AlreadySentDocumentTile {...defaultProps} />)

    const index = wrapper.find(`[data-test-id="already-sent-doc-index-${defaultProps.document.id}"]`)
    expect(index.props().children).toEqual(20)
  })
})
