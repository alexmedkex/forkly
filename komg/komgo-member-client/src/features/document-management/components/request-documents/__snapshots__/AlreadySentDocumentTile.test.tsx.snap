// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlreadySentDocumentTile should match default snapshot 1`] = `
.c4 {
  text-align: center;
  cursor: pointer;
  color: #5700b5;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c1 {
  height: 21px;
  width: 5px;
  color: #5d768f;
  font-family: Helvetica;
  font-size: 14px;
  line-height: 21px;
}

.c3 {
  margin: 5px;
}

.c5 {
  margin: 5px;
  text-align: right;
}

.c0 {
  padding-left: 15px;
  display: grid;
  grid-template-columns: 5px [index] auto [panel-item];
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  display: grid;
  grid-template-columns: auto [panel-item] auto [panel-item] auto [panel-item];
  border: 1px solid #e8eef3;
  border-radius: 4px;
  margin: 5px 5px 5px 15px;
  padding-left: 12px;
}

<div
  className="c0"
>
  <div
    className="c1"
    data-test-id="already-sent-doc-index-1"
  >
    20
  </div>
  <div
    className="c2"
  >
    <div
      className="c3"
    >
      <span
        className="c4"
        data-test-id="already-sent-doc-link-1"
        onClick={[Function]}
      >
        AML Letter - test
      </span>
    </div>
    <div
      className="c5"
    />
    <div
      className="c5"
      data-test-id="already-sent-doc-date-1"
    >
      17 Dec 95 - 3:24 AM
    </div>
  </div>
</div>
`;
